.github/workflows/build-package.yml.template

```diff
@@ -110,15 +110,16 @@ jobs:
         run: |
```

Comment:
we *also* need to release in the old location for backwards compatibility. I'd just copy-paste these two blocks below or something. Think about what happens if one succeeds and the other fails though: will we correctly retry?

Reply:
> we *also* need to release in the old location for backwards compatibility.

Oh I didn't think of that.
Having backwards compatibility is great, though I do wonder how long will we have to support to old release schema?

There are currently more than 8000 releases on GitHub, I am a bit worries about their growth.

Perhaps we should remove some really old versions, given that it's unlikely anybody wants them and if they really do, they could build these themselves?

> Think about what happens if one succeeds and the other fails though: will we correctly retry?

I changed build-version.sh to check for the new release schema but not the old one, so it would work if we upload the old schema first, then the new schema.

Reply:
> Having backwards compatibility is great, though I do wonder how long will we have to support to old release schema?

When we release the new quickinstall and binstall clients, we should make them report to the stats server differently or something? This assumes that the stats server is actually working at the moment, which I suspect it might not be.

> There are currently more than 8000 releases on GitHub, I am a bit worries about their growth.

Don't worry about it: rust command-line binaries are pretty small compared to GUI apps and games. If GitHub has a problem with it then they will tell us and we can find a better solution.

Reply:
```
$ curl 'https://warehouse-clerk-tmp.vercel.app/api/agents?year=2022&month=7'
{"cargo-binstall/0.11.1":4129,"cargo-binstall/0.7.0":4,"cargo-binstall/0.6.2":30,"cargo-binstall/0.9.1":78,"cargo-binstall/0.9.2":2189,"curl/7.79.1":1,"cargo-binstall/0.11.0":76,"cargo-quickinstall client (alsuren@gmail.com)":6652,"cargo-binstall/0.10.0":9250,"cargo-binstall/0.8.0":13}
```
We should probably just put the quickinstall version number in the user agent and that would be enough.

Unfortunately, it looks like 2022/8 is the month that I switched over to collecting stats per-day, and I didn't update https://github.com/alsuren/warehouse-clerk-tmp/blob/master/pages/api/agents.ts .

Also, get-stats.sh is returning {} for me. I wonder if I've hit a size limit on my redis instance or something.

Reply:
> > Having backwards compatibility is great, though I do wonder how long will we have to support to old release schema?
> 
> When we release the new quickinstall and binstall clients, we should make them report to the stats server differently or something? This assumes that the stats server is actually working at the moment, which I suspect it might not be.

I tried running the url in get-stats.sh directly on my local machine and it gives me an empty json.

Perhaps there's a token for this?
But it also doesn't give an error.

> > There are currently more than 8000 releases on GitHub, I am a bit worries about their growth.
> 
> Don't worry about it: rust command-line binaries are pretty small compared to GUI apps and games. If GitHub has a problem with it then they will tell us and we can find a better solution.

Thanks, I probably overthink this.

Reply:
> We should probably just put the quickinstall version number in the user agent and that would be enough.

I can do another PR for this.

> Also, get-stats.sh is returning {} for me. I wonder if I've hit a size limit on my redis instance or something.

Yeah I've experienced the same thing.

Reply:
I fixed /api/stats and /api/agents. I suspect they've both been broken since july last year and I never noticed. :(

Reply:
> I can do another PR for this.

I've created  #166 for this.